### DISCUSSION TITLE
v27.0 testing

### DISCUSSION THREAD

**Discussion by stratospher** [Bitshala/BitcoinCore-PR-Review-Club#65] [PR Reviews]
[Testing Guide: Bitcoin Core 27.0 Release Candidate](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Candidate-Testing-Guide)

- [ ] Preparation
- [ ] mempool.dat v1/v2 compatibility
- [ ] v2 Transport on by Default
- [ ] netinfo backward compatibility with pre-v26 nodes
- [ ] v3 Transaction Policy
- [ ] CoinGrinder coin selection algorithm
- [ ] migratewallet RPC is no longer experimental

if you're coming, please leave a comment below sharing:
1. which OS you're running v27.0 on?
2. have you done [Preparation](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Candidate-Testing-Guide#preparation) steps detailed in testing guide?
3. shoot questions regarding rest of the testing guide! (or) cool stuff you observed!

**Comment by spraveenitpro** [comment]
I will join: 👋🏼
1. Mac OS Sonama 14.2
2. No - Onway
3. Sure.

**Comment by naiyoma** [comment]
1. Ubuntu 23
2. Yes I have gone through the guide and completed the setup locally 

Observations
a) mempool.dat v1/v2 compatibility
v1 format
0000000 0001 0000 0000 0000                    
0000008
v2
hexdump -n8 $DATA_DIR_27/regtest/mempool.dat
0000000 0002 0000 0000 0000                    
0000008

I guess this would imply that the mempool data has increased in v2 though I'm not sure about that

b) v2 Transport on by Default
I thought that connecting to a v2 seed node would mean that all my peers would also be v2. However, I noticed that when I use bcli getpeerinfo, it outputs both v2 and v1 peers.
Something interesting that I observed is that v1 peers don't have session ids 

```
"connection_type": "outbound-full-relay",
    "transport_protocol_type": "v1",
    "session_id": ""
    
```

```
 "connection_type": "outbound-full-relay",
    "transport_protocol_type": "v2",
    "session_id": "b8739386b90b8dd7216152f2ad6b044535ceef7970cbe426e6756a404ebc214e"

```

**Comment by jkciw** [comment]
I will join. 
1. Macos Sonoma
2. Completed the prep

**Comment by ankuwu03** [comment]
1. Windows 11 (using WSL: Ubuntu 22.04.)
2. yeah done
3. I observed backward compatibility to be really interesting and am currently comparing it to above ones
Hope to have a good learning experience and hopefully contribute meaningfully :)
